extensions/amp-base-carousel/1.0/dimensions.js

```diff
@@ -0,0 +1,229 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {mod} from '../../../src/utils/math';
+
+/**
+ * @enum {number}
+ */
+export const Axis = {
+  X: 0,
+  Y: 1,
+};
+
+/**
+ * @enum {string}
+ */
+export const Alignment = {
+  START: 'start',
+  CENTER: 'center',
+};
+
+/**
+ * @param {!Axis} axis The Axis to get the Dimension for.
+ * @param {*} el The Element to get the Dimension For.
+ * @return {!DimensionDef} The dimension for the Element along the given Axis.
```

Comment:
You may need to copy the `DimensionDef` declaration here. Not 100%, but I don't see how it'd get picked up from 0.1 here. Unless it's externed there?

Reply:
When I copy the declaration here, I get: 
`Duplicate typedef name found: DimensionDef. Another typedef with the same name is found in extensions/amp-base-carousel/0.1/dimensions.js. Suggestion: Add an infix version indicator like MyType_0_1_Def.eslint(local/no-duplicate-name-typedef).`

Reply:
For now I have declared a unique `BaseCarouselDimensionDef` here.